### ISSUE TITLE
[Feature] support  paimon catalog

### ISSUE THREAD

**Issue by melin** [apache/incubator-paimon#841]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

See iceberg. Custom catalogs are supported. For example, users can customize catalogs based on jdbc. Table metadata can be stored in a relational database, independent of hive meta.

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by JingsongLi** [comment]
> See iceberg. Custom catalogs are supported. For example, users can customize catalogs based on jdbc. Table metadata can be stored in a relational database, independent of hive meta.
> 
> https://iceberg.apache.org/docs/latest/jdbc/ <img alt="image" width="1171" src="https://user-images.githubusercontent.com/1145830/230308479-c87fcb77-10cf-4bff-8bd2-19f23176b2eb.png">

Hi @melin thanks for reporting, is this requirement comes from your company production env?

**Comment by melin** [comment]
Yes, metadata storage does not depend on hms. can customize the metadata storage mode, which is suitable for more scenarios, such as cloud deployment.

**Comment by s7monk** [comment]
Can you implement this feature, if no one has done it yet, I am very interested in this, I can try to complete it

**Comment by s7monk** [comment]
Are you going to implement this feature, if no one has done it yet, I am very interested in this, I can try to complete it

**Comment by s7monk** [comment]
Are you going to implement this feature? If no one has done it yet, I am very interested in this, I can try to complete it

**Comment by s7monk** [comment]
I think we also need jdbc catalog or user-defined catalog. We currently rely on hms, but we have the idea of breaking away from hive

**Comment by melin** [comment]
It is currently implemented based on iceberg. iceberg catalog provides a fileio interface for more customized acceleration.
hms is the metadata storage system of hadoop data warehouse, which is no longer satisfied with the data lake metadata storage system.
iceberg rest catalog is a more scalable approach, leaving the implementation to the user

**Comment by s7monk** [comment]
> It is currently implemented based on iceberg. iceberg catalog provides a fileio interface for more customized acceleration. hms is the metadata storage system of hadoop data warehouse, which is no longer satisfied with the data lake metadata storage system. iceberg rest catalog is a more scalable approach, leaving the implementation to the user

Yes, if it is on the cloud, I think hive may not be very necessary, so we need to give users more choices

**Comment by s7monk** [comment]
Hi, what's the progress?

**Comment by zhangjun0x01** [comment]
+1 for custom catalog

1. Object storage cannot guarantee the atomic rename。
2. hive metastore rely on hms and mysql , it may become a bottleneck
3. business requirements: one tenant may have many sub account , hope to achieve resource isolation by catalog
